main.py

```diff
@@ -66,6 +66,7 @@ def __init__(self, robot):
     self.smile_cascade = cv2.CascadeClassifier(
         'haarcascades/haarcascade_smile.xml')
     self.robot = robot
+    self.last_action_time = time.time()
```

Comment:
I already calibrate the motors when they're constructed. I suppose you could do that here instead, but that would make it possible to start detecting at a funny angle before calibrating.